### PR TITLE
docs: Update LayoutLMv3 model card with standardized format and impro…

### PR DESCRIPTION
**Pull request by carrycooldude**
# Update LayoutLMv3 Model Card Documentation

This PR updates the LayoutLMv3 model card documentation to follow the standardized format as requested in #36979. The changes improve the documentation's clarity and usability while maintaining consistency with other model cards in the repository.

# What does this PR do?

This PR enhances the LayoutLMv3 model card documentation by:
- Adding badges for framework support (PyTorch, TensorFlow, Flax) and optimizations (Flash Attention, SDPA)
- Reorganizing code examples into clear sections:
  - Quick Start (basic usage)
  - Pipeline API examples
  - AutoModel examples
  - transformers-cli examples
- Adding quantization examples for large models (8-bit and 4-bit)
- Adding attention visualization examples using AttentionMaskVisualizer
- Maintaining existing functionality while improving documentation structure

The changes make the documentation more accessible and provide ready-to-use examples for different use cases, following the standardized format used in other model cards like Gemma 3, PaliGemma, and ViT.

#36979

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

Since this is a documentation update for a vision-language model, I would suggest tagging:
- @amyeroberts (vision models)
- @stevhliu (documentation)

### CODE REVIEW

File: docs/source/en/model_doc/layoutlmv3.md [hunk 1]
```diff
@@ -14,24 +14,150 @@ rendered properly in your Markdown viewer.
 
 -->
 
+[![PyTorch](https://img.shields.io/badge/PyTorch-1.12+-blue.svg)](https://pytorch.org/get-started/locally/)
```
Comment (stevhliu) [inline docs/source/en/model_doc/layoutlmv3.md:17]: Please style these with the `<div>` tags. You can copy it from one of the existing updated model cards on `main` like Gemma 3.

File: docs/source/en/model_doc/layoutlmv3.md [hunk 2]
```diff
@@ -14,24 +14,150 @@ rendered properly in your Markdown viewer.
...
+[![TensorFlow](https://img.shields.io/badge/TensorFlow-2.12+-orange.svg)](https://www.tensorflow.org/install)
+[![Flax](https://img.shields.io/badge/Flax-0.6+-yellow.svg)](https://flax.readthedocs.io/en/latest/installation.html)
+[![Safetensors](https://img.shields.io/badge/Safetensors-0.3+-green.svg)](https://huggingface.co/docs/safetensors/installation)
+[![Flash Attention](https://img.shields.io/badge/Flash%20Attention-2.0+-blue.svg)](https://github.com/Dao-AILab/flash-attention)
+[![SDPA](https://img.shields.io/badge/SDPA-2.0+-blue.svg)](https://pytorch.org/docs/stable/generated/torch.nn.functional.scaled_dot_product_attention.html)
+
 # LayoutLMv3
 
-## Overview
+LayoutLMv3 is a powerful multimodal transformer model designed specifically for Document AI tasks. What makes it unique is its unified approach to handling both text and images in documents, using a simple yet effective architecture that combines patch embeddings with transformer layers. Unlike its predecessor LayoutLMv2, it uses a more streamlined approach with patch embeddings (similar to ViT) instead of a CNN backbone.
```
Comment (stevhliu) [inline docs/source/en/model_doc/layoutlmv3.md:26]:
```suggestion
[LayoutLMv3](https://huggingface.co/papers/2204.08387) is a multimodal transformer model designed specifically for Document AI tasks. It unites the pretraining objective for text and images, masked language and masked image modeling, and also includes a word-patch alignment objective for even stronger text and image alignment. The model architecture is also unified and uses a more streamlined approach with patch embeddings (similar to [ViT](./vit)) instead of a CNN backbone.
```

File: docs/source/en/model_doc/layoutlmv3.md [hunk 3]
```diff
@@ -14,24 +14,150 @@ rendered properly in your Markdown viewer.
...
+
+The model is pre-trained on three key objectives:
+1. Masked Language Modeling (MLM) for text understanding
+2. Masked Image Modeling (MIM) for visual understanding
+3. Word-Patch Alignment (WPA) for learning cross-modal relationships
+
+This unified architecture and training approach makes LayoutLMv3 particularly effective for both text-centric tasks (like form understanding and receipt analysis) and image-centric tasks (like document classification and layout analysis).
+
+[Paper](https://arxiv.org/abs/2204.08387) | [Official Checkpoints](https://huggingface.co/microsoft/layoutlmv3-base)
```
Comment (stevhliu) [inline docs/source/en/model_doc/layoutlmv3.md:35]:
```suggestion
This unified approach makes LayoutLMv3 especially effective for text-centric tasks (form understanding and receipt analysis) and image-centric tasks (document classification and layout analysis).

You can find all the original LayoutLMv3 checkpoints under the [LayoutLM](https://huggingface.co/collections/microsoft/layoutlm-6564539601de72cb631d0902) collection.
```

File: docs/source/en/model_doc/layoutlmv3.md [hunk 4]
```diff
@@ -14,24 +14,150 @@ rendered properly in your Markdown viewer.
...
+
+<Tip>
+Click on the right sidebar for more examples of how to use the model for different tasks!
+</Tip>
```
Comment (stevhliu) [inline docs/source/en/model_doc/layoutlmv3.md:39]:
```suggestion
> [!TIP]
> Click on the LayoutLMv3 models in the right sidebar for more examples of how to apply LayoutLMv3 to different vision and language tasks. 
```

File: docs/source/en/model_doc/layoutlmv3.md [hunk 5]
```diff
@@ -14,24 +14,150 @@ rendered properly in your Markdown viewer.
...
+
+## Quick Start
```
Comment (stevhliu) [inline docs/source/en/model_doc/layoutlmv3.md:41]: Replace these headers (quick start, using the pipeline, automodel, etc.) with the `<hfoptions><hfoption>...</hfoption></hfoptions>` tags

File: docs/source/en/model_doc/layoutlmv3.md [hunk 6]
```diff
@@ -14,24 +14,150 @@ rendered properly in your Markdown viewer.
...
+
+Here's a quick example of how to use LayoutLMv3 for document understanding:
```
Comment (stevhliu) [inline docs/source/en/model_doc/layoutlmv3.md:43]:
```suggestion
The example below demonstrates how to answer a question based on an image with [`Pipeline`] or the [`AutoModel`] class.
```

File: docs/source/en/model_doc/layoutlmv3.md [hunk 7]
```diff
@@ -14,24 +14,150 @@ rendered properly in your Markdown viewer.
...
+
+```python
```
Comment (stevhliu) [inline docs/source/en/model_doc/layoutlmv3.md:45]: We can remove this code snippet

File: docs/source/en/model_doc/layoutlmv3.md [hunk 8]
```diff
@@ -14,24 +14,150 @@ rendered properly in your Markdown viewer.
...
+encoding = processor(image, return_tensors="pt")
+
+# Get predictions
+outputs = model(**encoding)
+predictions = torch.nn.functional.softmax(outputs.logits, dim=-1)
+```
+
+## Using the Pipeline
+
+The easiest way to use LayoutLMv3 is through the pipeline API:
```
Comment (stevhliu) [inline docs/source/en/model_doc/layoutlmv3.md:65]: The Pipeline example should be inside a `<hfoption>` block, and lets just focus on question answering with a real receipt image.

File: docs/source/en/model_doc/layoutlmv3.md [hunk 9]
```diff
@@ -14,24 +14,150 @@ rendered properly in your Markdown viewer.
...
+# For document classification
+classifier = pipeline("document-classification", model="microsoft/layoutlmv3-base")
+result = classifier("document.jpg")
+
+# For token classification (e.g., form understanding)
+token_classifier = pipeline("token-classification", model="microsoft/layoutlmv3-base")
+result = token_classifier("form.jpg")
+
+# For question answering
+qa = pipeline("document-question-answering", model="microsoft/layoutlmv3-base")
```
Comment (stevhliu) [inline docs/source/en/model_doc/layoutlmv3.md:79]:
```suggestion
qa = pipeline(task="document-question-answering", model="microsoft/layoutlmv3-base", torch_dtype=torch.bfloat16, device=0)
```

File: docs/source/en/model_doc/layoutlmv3.md [hunk 10]
```diff
@@ -14,24 +14,150 @@ rendered properly in your Markdown viewer.
...
+result = qa(question="What is the total amount?", image="receipt.jpg")
+```
 
-The LayoutLMv3 model was proposed in [LayoutLMv3: Pre-training for Document AI with Unified Text and Image Masking](https://arxiv.org/abs/2204.08387) by Yupan Huang, Tengchao Lv, Lei Cui, Yutong Lu, Furu Wei.
-LayoutLMv3 simplifies [LayoutLMv2](layoutlmv2) by using patch embeddings (as in [ViT](vit)) instead of leveraging a CNN backbone, and pre-trains the model on 3 objectives: masked language modeling (MLM), masked image modeling (MIM)
-and word-patch alignment (WPA).
+## Using AutoModel
```
Comment (stevhliu) [inline docs/source/en/model_doc/layoutlmv3.md:83]: This should be contained in an `<hfoption>` block as well

File: docs/source/en/model_doc/layoutlmv3.md [hunk 11]
```diff
@@ -14,24 +14,150 @@ rendered properly in your Markdown viewer.
...
+# Process inputs
+image = Image.open("document.jpg").convert("RGB")
+encoding = processor(image, return_tensors="pt")
 
-This model was contributed by [nielsr](https://huggingface.co/nielsr). The TensorFlow version of this model was added by [chriskoo](https://huggingface.co/chriskoo), [tokec](https://huggingface.co/tokec), and [lre](https://huggingface.co/lre). The original code can be found [here](https://github.com/microsoft/unilm/tree/master/layoutlmv3).
+# Get predictions
+outputs = model(**encoding)
+```
+
+## Using transformers-cli
```
Comment (stevhliu) [inline docs/source/en/model_doc/layoutlmv3.md:102]: We can remove this since `transformers-cli` doesn't support image inputs

File: docs/source/en/model_doc/layoutlmv3.md [hunk 12]
```diff
@@ -14,24 +14,150 @@ rendered properly in your Markdown viewer.
...
+
+For quick inference from the command line:
+
+```bash
+transformers-cli document-classification "document.jpg" --model microsoft/layoutlmv3-base
+```
+
+## Quantization
+
+For large models, you can use quantization to reduce memory usage:
```
Comment (stevhliu) [inline docs/source/en/model_doc/layoutlmv3.md:112]: Update the code example below accordingly

```suggestion
Quantization reduces the memory burden of large models by representing the weights in a lower precision. Refer to the [Quantization](https://huggingface.co/docs/transformers/main/en/quantization/overview) overview for more available quantization backends.

The example below uses [torchao](https://huggingface.co/docs/transformers/main/en/quantization/torchao) to only quantize the weights to int4.
```

File: docs/source/en/model_doc/layoutlmv3.md [hunk 13]
```diff
@@ -14,24 +14,150 @@ rendered properly in your Markdown viewer.
...
+
+# Or with 4-bit quantization
+model = LayoutLMv3ForSequenceClassification.from_pretrained(
+    "microsoft/layoutlmv3-base",
+    load_in_4bit=True,
+    device_map="auto"
+)
+```
+
+## Attention Visualization
```
Comment (stevhliu) [inline docs/source/en/model_doc/layoutlmv3.md:133]: I don't think the attention visualizer is supported for this model yet

File: docs/source/en/model_doc/layoutlmv3.md [hunk 14]
```diff
@@ -74,81 +200,66 @@ LayoutLMv3 is nearly identical to LayoutLMv2, so we've also included LayoutLMv2
 **Document question answering**
 - [Document question answering task guide](../tasks/document_question_answering)
 
-## LayoutLMv3Config
+## Model Details
```
Comment (stevhliu) [inline docs/source/en/model_doc/layoutlmv3.md:203]: The rest of these changes should be reverted

File: docs/source/en/model_doc/layoutlmv3.md [hunk 15]
```diff
@@ -14,24 +14,150 @@ rendered properly in your Markdown viewer.
...
+# Visualize attention
+visualizer = AttentionMaskVisualizer()
+visualizer.visualize_attention(
+    image,
+    outputs.attentions[-1][0],  # Last layer attention
+    processor.tokenizer
+)
+```
 
 ## Usage tips
```
Comment (stevhliu) [inline docs/source/en/model_doc/layoutlmv3.md:162]: Add a `## Notes` section here about [data processing](https://huggingface.co/docs/transformers/main/en/model_doc/layoutlmv3#usage-tips)

**Review by stevhliu — COMMENTED** [review]
Thanks, this is a good start! Please refer to the [Gemma 3 docs](https://huggingface.co/docs/transformers/main/en/model_doc/gemma3) see how to standardize this doc 🤗

**Comment by carrycooldude** [comment]
@stevhliu , have a look on this

**Comment by stevhliu** [comment]
Hey, it is still a bit off! For example:

- the badges look different and should be aligned on the right
- the code examples should be inside `<hfoption>` blocks so users can easily toggle between `Pipeline` and `AutoModel`
- the Resources section should be removed

I suggest taking a look at the [Gemma 3](https://github.com/huggingface/transformers/blob/main/docs/source/en/model_doc/gemma3.md?plain=1) model card again and trying to align your model card with it as much as possible!

File: docs/source/en/model_doc/layoutlmv3.md (hunk 2 above)
Comment (stevhliu) [reply docs/source/en/model_doc/layoutlmv3.md:26]: Not fully resolved yet, missing link to the model

File: docs/source/en/model_doc/layoutlmv3.md [hunk 16]
```diff
@@ -14,141 +14,172 @@ rendered properly in your Markdown viewer.
...
+[Paper](https://arxiv.org/abs/2204.08387) | [Official Checkpoints](https://huggingface.co/microsoft/layoutlmv3-base)
```
Comment (stevhliu) [inline docs/source/en/model_doc/layoutlmv3.md:35]:
```suggestion
You can find all the original LayoutLMv3 checkpoints under the [LayoutLM](https://huggingface.co/collections/microsoft/layoutlm-6564539601de72cb631d0902) collection.
```

File: docs/source/en/model_doc/layoutlmv3.md (hunk 4 above)
Comment (stevhliu) [reply docs/source/en/model_doc/layoutlmv3.md]: Not resolved yet either

File: docs/source/en/model_doc/layoutlmv3.md (hunk 14 above)
Comment (stevhliu) [reply docs/source/en/model_doc/layoutlmv3.md]: Not resolved yet, the `## Model Details` is still there as are the changes to the header levels of the LayoutLMv3 classes

File: docs/source/en/model_doc/layoutlmv3.md (hunk 12 above)
Comment (stevhliu) [reply docs/source/en/model_doc/layoutlmv3.md]: Not resolved yet. You only need to show quantization for either 8 or 4-bits instead of both. Also the code for quantizing the model is incorrect.

File: docs/source/en/model_doc/layoutlmv3.md (hunk 11 above)
Comment (stevhliu) [reply docs/source/en/model_doc/layoutlmv3.md]: Unresolved

File: docs/source/en/model_doc/layoutlmv3.md (hunk 9 above)
Comment (stevhliu) [reply docs/source/en/model_doc/layoutlmv3.md:57]: Unresolved

File: docs/source/en/model_doc/layoutlmv3.md (hunk 8 above)
Comment (stevhliu) [reply docs/source/en/model_doc/layoutlmv3.md]: Unresolved as there are still other examples here besides question answering

File: docs/source/en/model_doc/layoutlmv3.md (hunk 6 above)
Comment (stevhliu) [reply docs/source/en/model_doc/layoutlmv3.md]: Unresolved

**Review by stevhliu — COMMENTED** [review]
There are a lot of unresolved changes, so please don't mark them as resolved 😅

**Comment by stevhliu** [comment]
We'll need to update the badges to include FlashAttention and the code examples to include SDPA once #35469 is merged!

**Comment by carrycooldude** [comment]
> We'll need to update the badges to include FlashAttention and the code examples to include SDPA once #35469 is merged!

Sure , Will see to that too
